native-addons-are-close-to-being-on-par-with-js-modules.md

```diff
@@ -0,0 +1,113 @@
+# Native Addons Are Close To Being On Par With JS Modules
+
+Native addons have been a part of Node.js since the very beginning. So have JS modules 😉 So, what does it mean that they are now close to being "on-par"? Why only now?
+
+Let's first look at the similarities between native adons and JS modules. Both can be `require()`-ed:
+
+```JS
+// A JS module can be loaded from a package:
+const jsModulePackage = require('my-js-module');
+// It can also be loaded from a specific file:
+const jsModuleFile = require('./my-js-module.js');
+
+// A native addon can also be loaded from a file:
+const nativeAddonFile = require('./build/Release/my-native-addon.node');
+```
+
+Both return a JavaScript object containing useful functionality which would otherwise be completely unavailable, or which would otherwise take a long time to implement.
+
+Nevertheless, the similarities end there. Let's now contrast the two types of modules:
+
+|  |  | JavaScript modules | Native addons |
+|--- |--- |--- |--- |
+| 1. | ... need to be compiled. | No | Not if there are pre-builds |
+| 2. | ... will work on all platforms and architectures. | Yes | Yes if there are pre-builds |
+| 3. | ... will work with all current and subsequent Node.js versions once written. | Yes | No |
+| 4. | ... can be loaded multiple times. | Yes | No |
+| 5. | ... are thread-safe if not explicitly making use of threading infrastructure. | Yes | No |
+
+These differences have created a sizeable maintenance burden for those who choose to publish native addons, and for those whose deployed applications include native addons.
+
+### 1. Compilation
+This will always be a fundamental difference between native addons and JavaScript modules. The process whereby source code written in C or C++ (or even a different language such as Rust or Go) gets bundled into a native addon will always be an extra step in addition to publishing an npm package which users can consume. In fact, if the maintainer does not take the time to provide pre-built packages by using tools such as [node-pre-gyp][] or [prebuild][], then those who would depend on the native addon have to install a compiler and build tool chain to allow the native addon package to build on their system. Nevertheless, npm packages providing pre-built native addons can make it appear as though a native addon can be installed as easily as a JavaScript module.
```

Comment:
Maybe changing the order native JavaScript modules and addons, just following the table? And  this order or the other one could be check through the entire material.